include/libtorrent/enum_net.hpp

```diff
@@ -94,7 +94,9 @@ namespace libtorrent {
 	TORRENT_EXTRA_EXPORT bool in_local_network(std::vector<ip_interface> const& net
 		, address const& addr);
 
-	TORRENT_EXTRA_EXPORT address get_default_gateway(io_service& ios, error_code& ec);
+	// returns the first default gateway found if device is empty
+	TORRENT_EXTRA_EXPORT address get_default_gateway(io_service& ios
+		, string_view device, bool v6, error_code& ec);
```

Comment:
I wonder if it would be better to pass in the `boost::asio::udp::v6` or `boost::asio::udp::v4` instead of this `bool`. The `udp` or `tcp` part is a bit odd I suppose, but the call site would probably be easier to understand than `true` or `false`

Reply:
In this case it would be pretty awkward because both the caller and callee are dealing with `address`es so it would mean converting bool -> protocol -> bool. I think it's clear enough what's going on at the one-and-only call site of this function.